src/nsidc/metgen/metgen.py

```diff
@@ -345,12 +356,96 @@ def null_operation(configuration: config.Config, granule: Granule) -> Granule:
     return granule
 
 
+def edl_login(environment):
+    """
+    Authenticate with Earthdata using user name and password retrieved
+    from environment variables.
+    """
+
+    logger = logging.getLogger(constants.ROOT_LOGGER)
+    try:
+        earthaccess.login(
+            strategy="environment", system=getattr(earthaccess, environment)
+        )
+        auth = True
+    except LoginStrategyUnavailable:
+        logger.info(
+            "Environment variables EARTHDATA_USERNAME and EARTHDATA_PASSWORD \
+are missing."
+        )
+        auth = False
+    except LoginAttemptFailure as e:
+        logger.info(e)
+        auth = False
+
+    return auth
+
+
+def ummc_content(umm: list, key: str):
+    val = None
+
+    if not umm:
+        return val
+
+    # if more than one response, log error, use first one?
+    if not isinstance(umm[0], dict):
+        return val
+
+    logger = logging.getLogger(constants.ROOT_LOGGER)
+    try:
+        val = umm[0]["umm"][key]
+        logger.info(f"Found {key} information in umm-c response from CMR.")
+    except KeyError:
+        logger.info(f"No {key} information in umm-c response from CMR.")
+
+    return val
+
+
+def edl_environment(environment):
+    """
+    Map a cumulus ingest environment to the environment string needed for
+    Earthdata login
+    """
+    if environment.lower() != "prod":
+        environment = "uat"
+
+    return environment.upper()
+
+
+def edl_provider(environment):
+    return "NSIDC_PROD" if environment.lower() == "prod" else "NSIDC_CUAT"
```

Comment:
Should we put the provider name in config or is this considered a 'constant' that depends on the environment?